### PR TITLE
Rust style conventions for connstring

### PR DESCRIPTION
**Pull request by kyeah**
Stylistic changes recommended by Steve in #39.

### CODE REVIEW

File: src/client/connstring.rs [hunk 1]
```diff
@@ -22,7 +21,7 @@ impl Host {
         }
     }
 
-    fn new_ipc(ipc: String) -> Host {
+    fn with_ipc(ipc: String) -> Host {
```
Comment (vkarpov15) [inline src/client/connstring.rs]: Minor nitpick: I think what steve meant was `new_with_ipc` is more idiomatic rust.

File: src/client/connstring.rs [hunk 2]
```diff
@@ -79,10 +70,10 @@ impl ConnectionString {
     /// Creates a new ConnectionString for a single, unreplicated host.
     pub fn new(host_name: &str, port: u16) -> ConnectionString {
         let host = Host::new(host_name.to_owned(), port);
-        ConnectionString::new_from_host(host)
+        ConnectionString::with_host(host)
     }
 
-    fn new_from_host(host: Host) -> ConnectionString {
+    fn with_host(host: Host) -> ConnectionString {
```
Comment (vkarpov15) [inline src/client/connstring.rs]: Ditto, think this should be `new_with_host`

File: src/client/connstring.rs (hunk 1 above)
Comment (kyeah) [inline src/client/connstring.rs]: ack - fixing.

Comment (kyeah) [inline src/client/connstring.rs]: Actually, should be ```with_*```. std::vec::Vec provides a [```with_capacity``` constructor](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity).

**Pull request merged**